ambry-server/src/main/java/com/github/ambry/server/RepairRequestsDB.java

```diff
@@ -0,0 +1,268 @@
+    try {
+      PreparedStatement statement = connection.prepareStatement(INSERT_QUERY);
+      statement.setBytes(1, Base64.decodeBase64(record.getBlobId()));
+      statement.setInt(2, record.getPartitionId());
+      statement.setString(3, record.getSourceHostName());
+      statement.setInt(4, record.getSourceHostPort());
+      statement.setShort(5, record.getOperationType());
+      statement.setTimestamp(6, new Timestamp(record.getOperationTimeMs()));
+      statement.setShort(7, record.getLifeVersion());
+      if (record.getExpirationTimeMs() != Utils.Infinite_Time) {
+        statement.setTimestamp(8, new Timestamp(record.getExpirationTimeMs()));
+      } else {
+        statement.setTimestamp(8, null);
+      }
+      statement.executeUpdate();
+    } catch (SQLException e) {
+      logger.warn("failed to insert record to {} due to {}", dbEndpoint.getUrl(), e.getMessage());
+      throw e;
+    }
+  }
+
+  /**
+   * Select the records from one partition ordered by the operation time.
+   * @param partitionId partition id
+   * @return the oldest {@link RepairRequestRecord}s.
+   */
+  public List<RepairRequestRecord> getRepairRequests(int partitionId) throws SQLException {
+    // private static final String GET_QUERY = String.format(""
+    //    + "SELECT %s, %s, %s, %s, %s, %s, %s, %s "
+    //    + "FROM %s "
+    //    + "WHERE %s = ? "
+    //    + "ORDER BY %s ASC "
+    //    + "LIMIT 100",
+    //    BLOB_ID, PARTITION_ID, SOURCE_HOST_NAME, SOURCE_HOST_PORT, OPERATION_TYPE, OPERATION_TIME, LIFE_VERSION, EXPIRATION_TYPE,
+    //    REPAIR_REQUESTS_TABLE,
+    //    PARTITION_ID,
+    //    OPERATION_TIME);
+    try {
+      PreparedStatement statement = connection.prepareStatement(GET_QUERY);
+      statement.setInt(1, partitionId);
+      ResultSet resultSet = statement.executeQuery();
+      List<RepairRequestRecord> result = new ArrayList<>();
+      while (resultSet.next()) {
+        String blobId = Base64.encodeBase64URLSafeString(resultSet.getBytes(1));
+        // resultSet.getInt(2) is the partition id.
+        String sourceHostName = resultSet.getString(3);
+        int sourceHostPort = resultSet.getInt(4);
+        short operationType = resultSet.getShort(5);
+        Timestamp operationTime = resultSet.getTimestamp(6);
+        short lifeVersion = resultSet.getShort(7);
+        Timestamp expirationTime = resultSet.getTimestamp(8);
+        RepairRequestRecord record =
+            new RepairRequestRecord(blobId, partitionId, sourceHostName, sourceHostPort, operationType,
+                operationTime.getTime(), lifeVersion,
+                expirationTime != null ? expirationTime.getTime() : Utils.Infinite_Time);
+        result.add(record);
+      }
+      return result;
+    } catch (SQLException e) {
+      logger.warn("failed to get records from {} due to {}", dbEndpoint.getUrl(), e.getMessage());
```

Comment:
Should we use error instead of warnning when it's failed?
Also, can we have metrics with alert for all the requests since they are background jobs, so we can aware of something is wrong?

Reply:
Changed the warn to error.
You are right. We need add metrics as well.
I've added one TODO. Will clean them up one by one later.
// LOCAL_CONSISTENCY_TODO: add metrics for all the SQLException.